src/bmp/decoder.rs

```diff
@@ -770,9 +804,10 @@ impl<R: Read + Seek> BmpDecoder<R> {
                 BITMAPV4HEADER_SIZE => BMPHeaderType::V4,
                 BITMAPV5HEADER_SIZE => BMPHeaderType::V5,
                 _ => {
-                    return Err(ImageError::UnsupportedError(
+                    return Err(ImageError::Decoding(DecodingError::with_message(
+                        ImageFormat::Bmp.into(),
                         "Unsupported Bitmap Header".to_string(),
```

Comment:
Which should be used, `DecodingError` or `UnsupportedError`?

Possible candidate:

```rust
                    return Err(ImageError::Unsupported(
                        UnsupportedError::from_format_and_kind(
                            ImageFormat::Bmp.into(),
                            UnsupportedErrorKind::GenericFeature(
                                "Unsupported Bitmap Header".to_owned(),
                            ),
                        ),
                    ))
```

Reply:
Similar deal. There are technically more header sizes allowed corresponding to other header types which we do not precisely know (for the future and because there are unofficial extensions...). I think in this case it should always fail with `Unsupported` if the header size is larger than `BITMAPCOREHEADER_SIZE` and fail with `DecodingError` if it is smaller.